src/core/qgsfeaturefiltermodel.cpp

```diff
@@ -225,6 +225,14 @@ QVariant QgsFeatureFilterModel::data( const QModelIndex &index, int role ) const
 void QgsFeatureFilterModel::updateCompleter()
 {
   emit beginUpdate();
+
+  QgsFieldExpressionValuesGatherer *gatherer = qobject_cast<QgsFieldExpressionValuesGatherer *>( sender() );
+  if ( gatherer->wasCanceled() )
```

Comment:
Should we add this (just to be on the safe side)

```suggestion
  Q_ASSERT( gatherer );
  if ( gatherer->wasCanceled() )
```

Reply:
Sure!